File: observer/Customer.java
package observer;

public class Customer implements Subscriber {
    private String name;

    public Customer(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public void notify(String message) {
        System.out.println("Notifying user: " + name + " about: " + message);
    }
}

File: typescript/patterns/observer/customer.ts
import { Subscriber } from "./subscriber";

export class Customer implements Subscriber {
  constructor(private name: string) {}

  getName() {
    return this.name;
  }

  notify(message: string) {
    console.log(`Notifying user: ${this.name} about: ${message}`);
  }
}
